java/core/src/test/java/com/lancedb/lance/FragmentTest.java

```diff
@@ -55,6 +62,42 @@ void testFragmentScannerSchema() throws IOException, URISyntaxException {
     }
   }
 
+  @Test
+  void testFragmentScannerSubstrait() throws Exception {
+    String datasetPath = tempDir.resolve("fragment_substrait").toString();
+    try (BufferAllocator allocator = new RootAllocator()) {
+      TestUtils.SimpleTestDataset testDataset = new TestUtils.SimpleTestDataset(allocator, datasetPath);
+      testDataset.createEmptyDataset().close();
+      try (Dataset dataset = testDataset.write(1, 40)) {
+        var fragment = dataset.getFragments().get(0);
+        // This feature is not implemented: Unsupported function name: "equal:any_any", /Users/alluxio/alluxioFolder/lance/rust/lance-datafusion/src/expr.rs:496:9
+        String filterStatement = "id = 20";
+        String tableStatement =
+            "CREATE TABLE fragment_substrait (id INT NOT NULL, name VARCHAR(255) NOT NULL)";
+        SqlExpressionToSubstrait expressionToSubstrait = new SqlExpressionToSubstrait();
+        ExtendedExpression expression =
```

Comment:
This follows the example in https://arrow.apache.org/docs/java/substrait.html
leveraging https://github.com/substrait-io/substrait-java
@eddyxu

Reply:
And there is not that much java examples leveraging the subtraitFilter in ScanOptions,
and i could not find any example in Trino/Presto